content/solids/components/net.ts

```diff
@@ -0,0 +1,343 @@
+    previousBone: THREE.Bone,
+    faces: Polygon[],
+    hinges: Hinge[]
+): HingeBone[] {
+  const currentFace = faces[currentFaceIndex];
+  const hingesToProcess =
+    hingesForFace(currentFaceIndex, hinges)
+        .filter(hinge => !parentHinge(hinge, previousFaceIndex));
+
+  const nextFaces = hingesToProcess.map(hinge =>
+    otherFaceIndex(currentFaceIndex, hinge)
+  );
+
+  const boneHinges = nextFaces.reduce((acc: HingeBone[], faceIndex) => {
+    const nextFace = faces[faceIndex];
+    const hingeEdge = getCommonEdge(currentFace, nextFace);
+    const midpoint = hingeEdge.midpoint;
+
+    const angle = angleFromRadians(hingeEdge.angle);
+    const angleCoords = (new Point(1, 0)).rotate(angle.rad, new Point(0, 0));
+    const axis = new THREE.Vector3(round(angleCoords.x, 10), round(angleCoords.y, 10), 0);
+
+    const bone = new THREE.Bone();
+    bone.name = `${currentFaceIndex}-${faceIndex}`;
+    const parentPos = new THREE.Vector3();
+    previousBone.getWorldPosition(parentPos);
+    const parentPosPoint = new Point(parentPos.x, parentPos.y);
+    const diff = midpoint.subtract(parentPosPoint);
+    bone.position.set(diff.x, diff.y, 0);
+    previousBone.add(bone);
+
+    const verts: Point[] =
+      nextFace.points.filter(point =>
+        !point.equals(hingeEdge.p1) &&
+        !point.equals(hingeEdge.p2)
+      );
+
+    const hinge = hingeForFaces(faceIndex, currentFaceIndex, hinges);
+    const [_a, _b, maxAngleDeg] = hinge!;
+    const maxAngle = angleFromDegrees(maxAngleDeg).rad;
+
+    const boneHinge = {
+      bone,
+      verts,
+      axis,
+      maxAngle
+    };
+
+    const children = acc.concat(
+        boneHingesForFace(
+            faceIndex,
+            currentFaceIndex,
+            bone,
+            faces,
+            hinges
+        )
+    );
+    return [boneHinge, ...children];
+  },
+  []
+  );
+
+  return boneHinges;
+}
+/**
+  * Assumes that `a` and `b` are adjacent
+  */
+function getCommonEdge(a: Polygon, b: Polygon) {
+  return a.edges.filter(aEdge => b.edges.some(bEdge => aEdge.equals(bEdge)))[0];
+}
+
+function getBoneIndexFromPoint(point: Point, hingeBones: HingeBone[]) {
+  return hingeBones.findIndex(
+      hingeBone => hingeBone.verts.some(
+          vert => vert.equals(point)
+      )
+  );
+}
+
+function hingesForFace(faceIndex: number, hinges: Hinge[]) {
+  return hinges.filter(hinge =>
+    hinge[0] == faceIndex || hinge[1] == faceIndex
+  );
+}
+
+function hingeForFaces(a: number, b: number, hinges: Hinge[]) {
+  return hinges.find(hinge =>
+    (hinge[0] == a && hinge[1] == b) ||
+    (hinge[1] == a && hinge[0] == b)
+  );
+}
+
+function otherFaceIndex(currentFaceIndex: number, hinge: Hinge) {
+  if (currentFaceIndex == hinge[0]) return hinge[1];
+  else return hinge[0];
+}
+
+function mergePolys(polys: Polygon[]) {
+  const pointSet = new PointSet();
+  for (const poly of polys) {
+    pointSet.add(poly.points);
+  }
+  return new Polygon(...pointSet.points);
```

Comment:
I don't think we need `PointSet` as a separate class. Maybe just something like this:

``` ts
function mergePolys(polys: Polygon[]) {
  const points: Point[] = [];
  for (const poly of polys) {
    for (const point of poly.points) {
      if (points.every(p => !p.equals(point))) points.push(point)
    }
  }
  return new Polygon(...point.points);
}
```

At some point, we could even make this a static `Polygon.merge()` method in euclid.js